app/utils/auth.py

```
@@ -0,0 +1,277 @@
+            reason=INVALID_API_KEY,
+            log_message="Missing or empty API-KEY header"
+        )
+
+    try:
+        # Pass the API key directly to the authenticator
+        return await authenticator.authenticate(request, api_key_header)
+    except Exception as e:
+        raise UnauthorizedAccess(
+            reason=INVALID_API_KEY,
+            log_message=f"Authentication failed: {str(e)}"
+        )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)